net/acme-common/files/acme.sh

```diff
@@ -128,11 +134,16 @@ fi
 
 case $1 in
 get)
+	if [ -e "$run_dir"/lock ]; then
+		log err "Another ACME instance is already running."
+		exit 1
+	fi
+	touch "$run_dir/lock"
+	trap cleanup EXIT
```

Comment:
This is racy; two processes could run the check at the same time and both end up continuing. You'll have to use an atomic operation to actually create the lock; for shell scripts that's usually mkdir. See this example from sqm-scripts for how to do it right:

https://github.com/tohojo/sqm-scripts/blob/master/src/run-openwrt.sh#L80-L103

Reply:
Fixed. I decided to use flock. Seems it's designed specifically for this use case.

Reply:
I'm pretty sure this doesn't work the way you think it does; how did you test it?

Reply:
Sorry, I tested it in a small test script, but when tried to expand it in the real script, I got it wrong. It should be fixed.

I also fixed one more bug during testing, and took the liberty to add support for a new option. Let me know what you think.

Reply:
Yeah, that's better!